histogram.py

```diff
@@ -0,0 +1,318 @@
+        self.cryst_list = ['P', 'A', 'B', 'C', 'I', 'F', 'H', 'R']
+        self.list_data = []
+        self.data_included = []
+        for k, a_cryst in enumerate(self.cryst_list):
+            try:
+                self.list_data.append(data_to_histogram[a_cryst])
+            except KeyError:
+                self.list_data.append([])
+            self.data_included += self.list_data[k]
+
+        try:
+            self.data_excluded = []
+        except KeyError:
+            self.data_excluded = []
+
+        self.list_data.append(self.data_excluded)
+
+        self.max = max(self.data_included + self.data_excluded)
+        self.min = min(self.data_included + self.data_excluded)
+
+        self.color_exclude = 'lightgray'
+
+        self.list_colors = [colors[color] for color in self.cryst_list]
+        self.list_colors.append(self.color_exclude)
+
+        self.axs.set_title("Histogram of " + self.name)
+        self.axs.set_xlabel(self.xlabel)
+
+        _, _, self.patches = self.axs.hist(x=self.list_data, bins=self.bins,
+                                           density=1, stacked=True, alpha=0.9,
+                                           range=(self.min, self.max),
+                                           color=self.list_colors,
+                                           histtype='stepfilled')
+        # Draw the histogram
+        # histtype = 'stepfilled' because for the default type refreshing is
+        # very slow.
+        # patches is a list of 8 Polygons containing matrix N2 where
+        # y height is in the second column.
+        # Drawing the grid:
+        self.axs.grid(True)
+        self.xlim = self.axs.get_xlim()
+        self.current_xlim = self.axs.get_xlim()
+
+    def reset(self):
+        """Restore the initial settings.
+        """
+        self.was_clicked_before = False
+        self.range_green_space = [None, None]
+        self.current_xlim = self.xlim
+
+    def bool_crystal_exluded_green_space(self, data):
+        """Method for checking if data is in the selection and if the
+        cristal will be included in the selection or not.
+
+        Parametetrs
+        -----------
+
+        data : double
+
+            The value given to check if it belongs to the range of ​​interest.
+
+        Returns
+        -------
+
+        Boolean value.
+        """
+        if self.range_green_space[0] is None or\
+           self.range_green_space[1] is None:
+            return False
+        elif data < self.range_green_space[0] or\
+                data > self.range_green_space[1]:
+            return True
+
+        return False
+
+    def set_range_green_space(self, minimum, maximum):
+        """Setting the range of selection.
+
+        Parameters
+        ----------
+
+        minimum : double
+
+            Left position the range of ​​interest.
+        maximum : double
+
+            Right position the range of ​​interest.
+        """
+        self.range_green_space = [minimum, maximum]
```

Comment:
Isn't too important here, but in general using `property` or the `@property` decorator is a more pythonic way to achieve this setter/getter behaviour. This goes for anything from here (line 153) to the end of this file

Reply:
Done in: 6d059e09359705d6b89a4ec118bb70289dc203d5